### ISSUE TITLE
[Package Issue]: LLVM doesn't install the arm64 version (woa64) on Windows Arm64

### ISSUE THREAD

**Issue by cristianadam** [microsoft/winget-pkgs#95154]
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Installation issue.

### Brief description of your issue

On a Windows Arm64 I expect to get the native arm64 version of LLVM.

### Steps to reproduce

On a Windows Arm64 device:

```
winget install llvm
```

### Actual behavior

Installs the x64 version.

https://github.com/microsoft/winget-pkgs/blob/master/manifests/l/LLVM/LLVM/15.0.5/LLVM.LLVM.installer.yaml only has x86 and x64 versions. 

### Expected behavior

Install the arm64 version.

See See https://github.com/llvm/llvm-project/releases/download/llvmorg-15.0.6/LLVM-15.0.6-woa64.exe

### Environment

```shell
!
```

**Comment by stephengillie** [comment]
Similar behavior (x64 version installed on arm64 system) in this issue:
* https://github.com/microsoft/winget-pkgs/issues/94605

**Comment by Trenly** [comment]
Possibly related to https://github.com/microsoft/winget-cli/issues/2731

**Comment by cristianadam** [comment]
No, it doesn't have anything with the detection of the correct arm64 architecture, or if it's running with a x64 emulation.

I am running the arm64 cmd.exe and winget should pick up the arm64 version of LLVM and https://github.com/microsoft/winget-pkgs/pull/95156 is doing exactly what I expect! Thank you.

TotalCommander doesn't have an arm64 native package, just x86, x64 or both x86 and x64 in one package. Has nothing to do with arm64 🙂

**Comment by SCIEN-00** [comment]
still same issue remaining on my system

**Comment by ComplexPlane** [comment]
Any updates on this?

**Comment by SpecterShell** [comment]
> Any updates on this?

The upstream hasn't provided the arm64 installer for many versions, so it has been temporarily removed from the manifests to ensure the other users (x86 + x64 users) can still receive updates.

**Comment by wmmc88** [comment]
It seems like they have been publishing woa64.exe installers for a while again. I checked releases from the 20.x.x prereleases all the way back to the 19.1.1 release, and it seems like the all have woa installers. Is there any reason not to add the arm installers back to the manifests?

**Comment by SpecterShell** [comment]
> It seems like they have been publishing woa64.exe installers for a while again. I checked releases from the 20.x.x prereleases all the way back to the 19.1.1 release, and it seems like the all have woa installers. Is there any reason not to add the arm installers back to the manifests?

Those installers were added weeks after the release.

![Image](https://github.com/user-attachments/assets/d9617847-323d-4ede-95d6-eba6b975d9d7)

**Comment by pronebird** [comment]
Besides winget installing x64 on arm64 windows, it doesn't update environment either. But standalone installer does. Maybe that should be addressed to?

**Comment by SpecterShell** [comment]
> Besides winget installing x64 on arm64 windows, it doesn't update environment either. But standalone installer does. Maybe that should be addressed to?

https://github.com/microsoft/winget-pkgs/issues/11767

**Comment by mihailik** [comment]
<img width="1209" height="686" alt="Image" src="https://github.com/user-attachments/assets/8b3ef7e4-e7d6-49ed-9eac-5bc27e0b7327" />

Ever since Microsoft employees got MacBooks their contempt towards paying customers of their own Windows has 🚀 **rocketed!!!**

<br>

<br>

**You** have **been identified* as **the party** requiring a fix, for being too cheap to attempt Windows, all the cool boys are on MacOS or at very least WSL.

That's the message we're seeing across developer tools in Windows.

**Comment by SpecterShell** [comment]
> > Anyway I will update my automation to make it detect and add woa64 installers.
> 
> [@SpecterShell](https://github.com/SpecterShell), may we get any updates on this?

It is still true that every time a new version is released (on [GitHub](https://github.com/llvm/llvm-project/releases)), the Windows x86+x64 installers come after only one day, and the Windows arm64 (woa64) installer comes after one week.

I've already updated my automation, so it will push the new version to this repo with only x86+x64 installers (in case the arm64 installer never got released for weeks), and update it as soon as the arm64 installer is released. Here is an example: https://github.com/microsoft/winget-pkgs/pull/307807

**Comment by leejy12** [comment]
@SpecterShell Thank you for your work.

> it will push the new version to this repo with only x86+x64 installers (in case the arm64 installer never got released for weeks), and update it as soon as the arm64 installer is released.

I’m not sure if this is desired behavior, as running `winget upgrade LLVM.LLVM` on an ARM device during the period when the ARM64 installer is unavailable—but the x64 installer is—will silently install the x64 version.
